### PR TITLE
Copy of PR #385 - Dead letter queue

### PR DESCRIPTION
**Pull request by raghavdixit99**
# Update GitHub Actions Workflow

- **Purpose:**
 Update the Python testing GitHub Actions workflow to include additional Python versions and set the environment variable for testing.
- **Key Changes:**
  - Added Python versions 3.11 and 3.12 to the testing matrix.
  - Set the `ENVIRONMENT` environment variable to `test`.
- **Impact:**
 The updated workflow will now run tests against a wider range of Python versions and indicate that the tests are being run in a test environment.

> ✨ Generated with love by [Kaizen](https://cloudcode.ai) ❤️

<details>
<summary>Original Description</summary>
 
Agentops PR #385 - Dead letter queue
</details>

### CODE REVIEW

File: tests/core_manual_tests/http_client/dead_letter_queue.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+# wait for the JWT to expire
+time.sleep(3)
+
+# retry
+HttpClient()._retry_dlq_requests()
+session.end_session(end_state="Success")
+
+# check if the failed request is still in the DLQ
+assert dead_letter_queue.get_all() == []
```
[the remark this answers is not in the document]
Comment (aiswaryasankar) [reply tests/core_manual_tests/http_client/dead_letter_queue.py:57]: @entelligence.ai how does the DLQ request logic actually work?

[the remark this answers is not in the document]
Comment (aiswaryasankar) [reply tests/core_manual_tests/http_client/dead_letter_queue.py:57]: @EntelligenceAI how does the DLQ request logic work?

[the remark this answers is not in the document]
Comment (aiswaryasankar) [reply tests/core_manual_tests/http_client/dead_letter_queue.py:57]: @Entelligence.AI how does the DLQ work?

[the remark this answers is not in the document]
Comment (raghavdixit99) [reply tests/core_manual_tests/http_client/dead_letter_queue.py:57]: @Entelligence.AI how does the DLQ work?

[the remark this answers is not in the document]
Comment (raghavdixit99) [reply tests/core_manual_tests/http_client/dead_letter_queue.py:57]: @Entelligence.AI what is DLQ?

[the remark this answers is not in the document]
Comment (raghavdixit99) [reply tests/core_manual_tests/http_client/dead_letter_queue.py:57]: @Entelligence.AI what is DLQ and how does it work?

**Pull request closed without merging**